lib/internal/modules/cjs/loader.js

```diff
@@ -532,18 +550,50 @@ function resolveExportsTarget(pkgPath, target, subpath, basePath, mappingKey) {
     }
   } else if (Array.isArray(target)) {
     for (const targetValue of target) {
-      if (typeof targetValue !== 'string') continue;
+      if (Array.isArray(targetValue)) continue;
       try {
         return resolveExportsTarget(pkgPath, targetValue, subpath, basePath,
                                     mappingKey);
       } catch (e) {
         if (e.code !== 'MODULE_NOT_FOUND') throw e;
       }
     }
+  } else if (typeof target === 'object' && target !== null) {
+    if (experimentalRequireTarget &&
+        ObjectPrototype.hasOwnProperty(target, 'require')) {
+      try {
+        return resolveExportsTarget(pkgPath, target.require, subpath,
+                                    basePath, mappingKey);
+      } catch (e) {
+        if (e.code !== 'MODULE_NOT_FOUND') throw e;
+      }
+    }
+    if (ObjectPrototype.hasOwnProperty(target, 'node')) {
```

Comment:
Based on the consensus in the last meeting I was under the impression that we were going to only support the default target for now. TBH I would personally like us to spend a tiny bit more time discussing semantics of the `node` and `require` targets before doing either.

As such it might make more sense to have a different flag than `experimentalRequireTarget`

Reply:
`experimentalMultipleEntries`?

Reply:
One downside of not supporting `node` right now is that (imo) it would be dangerous to unflag with this feature but without a `node` guard. It would effectively force anybody trying out conditional exports to consider `node` the one true JS ("default") implementation and force them to treat `browser` (a much more wide-spread JS runtime) as the exceptional thing.

But we could always resolve that by unflagging without conditional exports, putting the whole feature behind its own flag.

Reply:
(I'm fine landing this PR with either variant. This is a forward-looking statement for unflagging `--experimental-modules` in 13.)

Reply:
@jkrems shipping with only default makes this feature essentially equivalent to current behavior and more or less sugar for the below.

```json
{
  "exports": {
    ".": [{browser: "./browser.mjs" }, "./index.mjs"]
  }
}
```

I'm happy with either all the current semantics behind a single flag or default only for that reason. Introducing `node` comes with it programming patterns and semantics I want to discuss through a big more before unflagging.

Reply:
Sure, we can consider the "node" condition in a follow-up. I've updated the PR to implement the baseline "default" condition with the "require" condition flagged.